File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :square_of_sum, :sum_of_squares, :difference

  def initialize(number = 1)
    @square_of_sum = (1..number).to_a.sum**2
    @sum_of_squares = (1..number).map { |n| n**2 }.sum
    @difference = square_of_sum - sum_of_squares
  end
end

File: typescript/difference-of-squares/difference-of-squares.ts
export default class Square {
    private _squareOfSum : number;
    private _sumOfSquares : number;
    private _difference : number;

    constructor(n: number) {
        this._squareOfSum = Math.pow((n*(n+1))/2, 2)
        this._sumOfSquares = (n*(2*n + 1)*(n+1)) / 6
        this._difference = this._squareOfSum - this._sumOfSquares;
    }

    public get squareOfSum() : number {
        return this._squareOfSum;
    }

    public get sumOfSquares() : number {
        return this._sumOfSquares;
    }

    public get difference() : number {
        return this._difference;
    }
}